### PR TITLE
New API + more

### PR DESCRIPTION
**Pull request by kossnocorp**
- New API:

  - Adopt atomic modules approach:

    ```
    └── which_os
        ├── index.js
        └── test.js
    ```

  - Make API flat: get rid of `sniffFns.browser` namespace.

  - Move unbinded functions to `sniffFns.lib` namespace.

  - Add primitive properties to the top level
    (`sniffFns.isIE = sniffFns.lib.isIE(navigator.userAgent)`)
  
    ```js
    sniffFns.isIE
    //=> true
    ```

  - Normalize CommonJS file structure:

    ```diff
    - const isChrome = require('sniff-fns/src/is_chrome_browser')
    + const isChrome = require('sniff-fns/is_chrome')
    ```

- New tests suite:

  - Replace Karma + Mocha with AVA

  - Add test coverage support

- Automated release process.

- A couple of bug fixes:

  - Fixed `isMac` so it won't detect iPad as a Mac

  - Fixed `isBlackberry` pattern to use actual BlackBerry patterns.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
+sudo: false
 language: node_js
 node_js:
-  - '0.11'
+  - 4.4
```
Comment (conf) [inline .travis.yml]: What about 6.x? :)

File: scripts/_lib/list_fns.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const path = require('path')
+const fs = require('fs')
+
+module.exports = function listFns () {
+  const files = fs.readdirSync(path.join(process.cwd(), 'src', 'unbinded'))
+  return files
+    .filter((file) => !file.startsWith('.'))
+    .map((file) => {
+      const unbindedPath = `./src/unbinded/${file}/index.js`
```
Comment (conf) [inline scripts/_lib/list_fns.js]: Should `unbindedPath` be `unboundPath`?

Comment (kossnocorp) [inline scripts/_lib/list_fns.js]: @conf good point 👍

File: .travis.yml (hunk 1 above)
Comment (kossnocorp) [inline .travis.yml]: @conf what about 6.x? 😆 Luckily shrinkwrap isn't used here.

Comment (conf) [inline .travis.yml]: I meant to add it to `travis.yml`, so tests would also run on Node 6.x.

Comment (kossnocorp) [inline .travis.yml]: @conf I know, but the code doesn't touch that could be broken on Node.js upgrade, so I don't see a point.

**Pull request merged**